Common/Product/SharedProject/MsBuildProjectElement.cs

```diff
@@ -183,14 +183,7 @@ public override bool Equals(object obj)
             var include1 = GetMetadata(ProjectFileConstants.Include);
             var include2 = msBuildProjElem.GetMetadata(ProjectFileConstants.Include);
 
-            // Unfortunately the checking for nulls have to be done again, since neither String.Equals nor String.Compare can handle nulls.
```

Comment:
aren't the null checks still required? Equals will return true if they are both null which was not the case previously